core/modules/replica/protocol.proto

```diff
@@ -759,3 +759,55 @@ message ProtocolFileResponse {
     required uint32 mtime = 3;
 
 }
+
+//////////////////////////////////////////////
+//         The IngestServer protocol        //
+//////////////////////////////////////////////
+
+/// Sent once by a client after establishing a connection with a server
+message ProtocolIngestHandshakeRequest {
+
+    /// An identifier of a super-transaction
+    required uint32 transaction_id = 1;
+
+    /// The base name (no chunk numbers or 'overlap' suffixes) of table to be loaded
+    required string table = 2;
+
+    /// The chunk number
+    required uint32 chunk = 3;
+
+    /// The flag indicating if this is the chunk 'overlap' table
+    required bool is_overlap = 4;
+
+    /// A character separating columns within a row
+    enum ColumnSeparator {
+        COMMA = 0;
+        TAB   = 1;
+    }
+    required ColumnSeparator column_separator = 5;
+}
+
+/// A message which is meant for sending rows to a server after receiving
+/// status READY_TO_READ_DATA. This message is sent once after each READY_TO_READ_DATA.
+/// When the client has no date then it should send the last message with
+/// zero number of rows. And the server should respond with FINISHED. Then both
```

Comment:
Hmm, from the code that I saw it looks like "last" flag can be set to true even if number of rows is non-zero?

Reply:
Fixed